lib/workers/repository/config-migration/branch/migrated-data.ts

```diff
@@ -1,15 +1,56 @@
 import detectIndent from 'detect-indent';
 import JSON5 from 'json5';
+import prettier from 'prettier';
 import { migrateConfig } from '../../../../config/migration';
 import { logger } from '../../../../logger';
 import { readLocalFile } from '../../../../util/fs';
+import { getFileList } from '../../../../util/git';
 import { detectRepoFileConfig } from '../../init/merge';
 
 export interface MigratedData {
   content: string;
   filename: string;
 }
 
+export async function applyPrettierFormatting(
+  content: string,
+  indent: string
+): Promise<string> {
+  const prettierConfigFilenames = [
+    '.prettierrc',
+    '.prettierrc.json',
+    '.prettierrc.yml',
+    '.prettierrc.yaml',
+    '.prettierrc.json5',
+    '.prettierrc.js',
+    '.prettierrc.cjs',
+    'prettier.config.js',
+    'prettier.config.cjs',
+    '.prettierrc.toml',
+  ];
+  let prettierExists = (await getFileList()).some((file) =>
+    prettierConfigFilenames.includes(file)
+  );
+  if (!prettierExists) {
+    try {
+      const packageJsonContent = await readLocalFile('package.json', 'utf8');
+      prettierExists ||=
+        packageJsonContent && JSON.parse(packageJsonContent).prettier;
+    } catch {
+      logger.warn('Invalid JSON found in package.json');
+    }
+  }
+
+  if (!prettierExists) {
+    return content;
+  }
+
+  return prettier.format(content, {
+    parser: 'json',
+    tabWidth: indent.length,
```

Comment:
What about `tabs` ?

Reply:
`ident` can be `\t`, that needs to be detected here. Does the `json` parser also support `JSON5`?

Reply:
if tabs: true it means that indentation will use tabs instead of spaces..I am confused if it should be true or not?

[frozen](https://prettier.io/docs/en/options.html#tabs)

Reply:
https://www.npmjs.com/package/detect-indent#user-content-api

We probably need to pass the result from `detectIdent` call.

Reply:
How should we handle no indentation?